src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1815,137 +2357,160 @@ void func_80A153FC(EnGo* this, PlayState* play) {
-            case ENGO_70_2:
-                return D_80A163DC;
-            case ENGO_70_3:
-                return D_80A163EC;
-            case ENGO_70_4:
-                return D_80A163FC;
-            case ENGO_70_5:
-                return D_80A163FC;
+    if (this->sleepState != ENGO_AWAKE) {
+        return sMsgScriptGoronSleeping; // Lullaby Hint
+    }
+
+    if (ENGO_GET_TYPE(&this->actor) == ENGO_STRETCHER) {
+        switch (ENGO_GET_SUBTYPE(&this->actor) % 6) {
+            case ENGO_STRETCHER_A:
+                return sMsgScriptGoronStretcherA;
+            case ENGO_STRETCHER_B: // Unused!
+                return sMsgScriptGoronStretcherB;
+            case ENGO_STRETCHER_C:
+                return sMsgScriptGoronStretcherC;
+            case ENGO_STRETCHER_D:
+                return sMsgScriptGoronStretcherD;
+            case ENGO_STRETCHER_HAMSTRINGSIT:
+                return sMsgScriptGoronStretcherHamstring;
+            case ENGO_STRETCHER_HAMSTRINGSTAND:
+                return sMsgScriptGoronStretcherHamstring;
         }
     }
 
-    switch (ENGO_GET_F(&this->actor)) {
-        case ENGO_F_3:
-            return D_80A16350;
-        case ENGO_F_4:
-            return D_80A16704[ENGO_GET_70(&this->actor)];
-        case ENGO_F_5:
-            return D_80A16208;
-        case ENGO_F_6:
-            return D_80A16254;
-        case ENGO_F_7:
-            return D_80A16210;
-        case ENGO_F_8:
-            return D_80A16280;
+    switch (ENGO_GET_TYPE(&this->actor)) {
+        case ENGO_GATEKEEPER:
+            return sMsgScriptGoronGatekeeper;
+        case ENGO_GRAVEBRO:
+            return sMsgScriptGraveBros[ENGO_GET_SUBTYPE(&this->actor)];
+        case ENGO_ASIDE_STORE:
+            return sMsgScriptGoronAsideStore;
+        case ENGO_ASIDE_ELDERSROOM:
+            return sMsgScriptGoronAsideEldersRoom;
+        case ENGO_ASIDE_ELDERSSON:
+            return sMsgScriptGoronAsideEldersSon;
+        case ENGO_PKEG_SELLER:
+            return sMsgScriptGoronPowderKegSeller;
         default:
-            return D_80A16208;
+            return sMsgScriptGoronAsideStore;
     }
 }
 
-void func_80A157C4(EnGo* this, PlayState* play) {
+/**
+ * Action function for dialog, movement due to dialog, and ending dialog
+ */
+void EnGo_Talk(EnGo* this, PlayState* play) {
     s32 pad;
-    Vec3f sp40;
-    Vec3f sp34;
-
-    if (!func_8010BF58(&this->actor, play, func_80A15684(this, play), this->unk_3D8, &this->unk_28C)) {
-        if ((ENGO_GET_F(&this->actor) != ENGO_F_1) && !(this->unk_390 & 0x200)) {
-            Math_Vec3f_Copy(&sp40, &this->unk_38C->world.pos);
-            Math_Vec3f_Copy(&sp34, &this->actor.world.pos);
-            Math_ApproachS(&this->actor.shape.rot.y, Math_Vec3f_Yaw(&sp34, &sp40), 4, 0x2AA8);
+    Vec3f targetPos;
+    Vec3f thisPos;
+
+    if (!func_8010BF58(&this->actor, play, EnGo_GetMessageScript(this, play), this->msgEventCb,
+                       &this->msgScriptResumePos)) {
```

Comment:
Is this confirmed? All the other functions that use this don't have this argument named.

Reply:
Discovered when understanding the purpose by peeking ahead at [unfinished work](https://github.com/Thar0/mm/blob/msgevent/src/code/z_msgevent.c#L1054).